### PR TITLE
DHE Speed Up

### PR DESCRIPTION
**Pull request by ejohnstown**
When loading DH domain parameters into a CTX, test the prime immediately. When loading them into a session, test the prime right before using it during the handshake. Sessions that get their prime from
their context do not need to test their prime. Added a function to disable testing the prime in a session. The goal is to speed up testing as every single test case loads DH parameters whether they are used or not.

### CODE REVIEW

File: examples/server/server.c [hunk 1]
```diff
@@ -670,6 +670,7 @@ static void Usage(void)
 #ifdef WOLFSSL_EARLY_DATA
     printf("%s", msg[++msgId]);     /* -0 */
 #endif
+    printf("-X          Disable DH Prime check\n");
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply examples/server/server.c]: Will do. I am going to add support on the client side as well.

**Comment by toddouska** [comment]
Which cases are still checked in the .conf files?

File: tests/test.conf [hunk 2]
```diff
@@ -1,2295 +1,2814 @@
...
 -l DHE-PSK-AES128-CBC-SHA256
+-2
 
 # client TLSv1.1 DHE-PSK-AES128-CBC-SHA256
 -s
 -v 2
 -l DHE-PSK-AES128-CBC-SHA256
+-2
+
+# server TLSv1.2 DHE-PSK-AES128-CBC-SHA256 (DHE prime test)
```
Comment (dgarske) [inline tests/test.conf]: @toddouska : The only DHE Prime Test is here. @ejohnstown : Should we test some other cipher suites as well?

Comment (ejohnstown) [reply tests/test.conf]: There are three cipher suites with DHE testing in test.conf: DHE-RSA-AES128-SHA, DHE-RSA-AES256-SHA256, and DHE-PSK-AES128-CBC-SHA256.

Comment (toddouska) [reply tests/test.conf]: ok, thanks

**Pull request merged**